trio/_core/tests/test_run.py

```diff
@@ -1861,7 +1866,7 @@ def handle(exc):
                 else:  # pragma: no cover
                     return exc
```

Comment:
Is the `catch` handler supposed to reraise the exception if it's not handled? You documented that it's equivalent to an `except*` block, but here you're returning `exc` rather than raising it.

Reply:
This was a leftover from the original test code which did return `None` or `exc`. I've removed that now.